### DISCUSSION TITLE
Is there a way to get the logfile into the action message in which the regex found a match?

### DISCUSSION THREAD

**Discussion by ghost** [fail2ban/fail2ban#3628] [Q&A] [answered]
Is there a way to get the logfile into the action message in which the regex found a match?  

Something like this would be good.  `"<matchedlogfile>"`
```
curl -sSf "https://an.abuse.domain/report" -H "Accept: application/json" -H "Key: <custom_apikey>" --data-urlencode "logfile=<matchedlogfile>" --data-urlencode "comment=<matches>" --data-urlencode "ip=<ip>" --data "categories=<custom_category>"
```

The function "_grep_logs" did not return what I wanted, since i am using a wildcard for the logfiles.
```
access_*.log *_access.log
```

The background to this is that a large number of domains are active on our servers and we want to determine which ones are most affected.

**Comment by sebres** [comment, marked as answer]
No, there is no way at the moment without grep, see https://github.com/fail2ban/fail2ban/issues/2393#issuecomment-491291061
Another possibility would be a single jail per logpath, but it'd be a bit ugly. 
However in sense of domain only correct choice, because by `maxretry` larger than 1 it would consider only attempts per domain.
Although it is rather point of view, just... if by single jail and `maxretry = 5`, it will find 5 messages of 5 attempts in 3 different logpath, which one would you see in the action? The last one (e. g. log of last attempt caused ban)?

> The background to this is that a large number of domains are active on our servers and we want to determine which ones are most affected.

Well that has nothing with a logfile, if the domain is also a part of logged message (or a part of logged URL).
This will be also the sole possibility if you would monitor systemd journal instead of log file.
For instance log-format of nginx can be configured to write domain to the log.
In this way you'd see the domain(s) in the comment (related to `"comment=<matches>"`).

In such a case the messages may look like that:
```
192.0.2.123 - - [10/Nov/2023:19:53:28 +0100] "domain1" "GET /admin/scripts/setup.php HTTP/1.1" 403 47 "-" "-" "-"
192.0.2.124 - - [10/Nov/2023:19:53:29 +0100] "domain2" "GET /admin/scripts/setup.php HTTP/1.1" 403 47 "-" "-" "-"
```
You can even capture them with a filter:
```ini
[Definition]
block = [^"]*
code = (?!40[14])[45]0\d
failregex = ^<ADDR> \- <F-USER>\S+</F-USER> (?:\[\] )?"<F-DOMAIN>[^"]+</F-DOMAIN>" "[A-Z]+ <block>[^"]*" <code>\b
datepattern = ^[^\[]*\[({DATE})
              {^LN-BEG}
```
PoC:
```
$ fail2ban-regex -o '<ip> | <F-DOMAIN>' /tmp/log filter
192.0.2.123 | domain1
192.0.2.124 | domain2
```
So you could use `"<F-DOMAIN>"` in the actions to obtain the last domain of banned ticket. Again - **the last domain**, that means here the domain of last attempt that caused the ban, also if previously it has 4 attempts with another domain.

**Reply by ghost** [reply]
Thank you very much, this was very helpful and informative.
